packages/design-picker/src/components/index.tsx

```diff
@@ -210,6 +210,88 @@ const DesignButtonContainer: React.FC< DesignButtonContainerProps > = ( {
 	);
 };
 
+interface GeneratedDesignPreviewProps {
+	design: Design;
+	locale: string;
+}
+
+const GeneratedDesignPreview: React.FC< GeneratedDesignPreviewProps > = ( { design, locale } ) => {
+	return (
+		<div className="generated-design-preview">
+			<div className="generated-design-preview__header">
+				<svg width="36" height="8">
+					<g>
+						<rect width="8" height="8" rx="4" />
+						<rect x="14" width="8" height="8" rx="4" />
+						<rect x="28" width="8" height="8" rx="4" />
+					</g>
+				</svg>
+			</div>
+			<div className="generated-design-preview__frame">
+				<DesignPreviewImage design={ design } locale={ locale } highRes />
+			</div>
+		</div>
+	);
+};
+
+export interface GeneratedDesignPickerProps {
+	designs: Design[];
+	locale: string;
+	categorization?: Categorization;
+	heading?: React.ReactElement;
+	viewMoreFooter?: React.ReactNode;
+}
+
+const GeneratedDesignPicker: React.FC< GeneratedDesignPickerProps > = ( {
+	designs,
+	locale,
+	categorization,
+	heading,
+	viewMoreFooter,
+} ) => {
+	const categories = useMemo( () => {
+		if ( ! categorization ) {
+			return [];
+		}
+
+		return categorization.categories.map( ( category ) => {
+			const design = designs.find( ( _design ) => _design.slug === category.slug );
+			if ( ! design ) {
+				return category;
+			}
+
+			return {
+				...category,
+				previewUrl: getDesignPreviewUrl( design, { language: locale } ),
+			};
+		} );
+	}, [ designs, categorization?.categories ] );
+
+	return (
+		<div className="generated-design-picker">
+			{ heading }
+			<div className="generated_design-picker__content">
+				{ categorization && (
+					<DesignPickerCategoryFilter
+						categories={ categories }
+						selectedCategory={ null }
+						recommendedCategorySlug={ null }
+						footer={ viewMoreFooter }
+						onSelect={ categorization.onSelect }
+					/>
+				) }
+				<div className="generated-design-picker__previews">
+					{ designs &&
+						designs.map( ( design ) => (
+							<GeneratedDesignPreview key={ design.slug } design={ design } locale={ locale } />
```

Comment:
I think the layout is a bit weird. In my imagination, we show the generated designs and the "View more options" button in the sidebar on the desktop and show them in the center of the content on the mobile. And we have another larger preview on the desktop and it would be hidden on the mobile.

However, we show the generated designs by making them categories but actually, they're not categories. Then, we render all of the designs and “View more options” button here on the mobile and only show one of them on the desktop. It's not intuitive 😂

Reply:
> the concern that the layout structure is not intuitive

Yes, that's what I mean

> Do you mean that on mobile resolution, we should show the thumbnails and hide the previews rather than the other way around?

Yeah, this would be more intuitive. As we have created a new component named `GeneratedDesignPicker`, I think we could probably decouple the layout structure from the original one.

Reply:
Makes sense!